File: src/java/CharacterRepetition.java
import java.util.*;

public class CharacterRepetition {
    public static HashMap<Character, Integer> getUniqueLettersRepetition(String myString) {
        HashMap<Character, Integer> uniqueLettersMap = new HashMap<>();

        for (int i = 0; i < myString.length(); i++) {
            char currentLetter = myString.charAt(i);
            if (!uniqueLettersMap.containsKey(currentLetter)) {
                uniqueLettersMap.put(currentLetter, 1);
            } else {
                int repetitionOfCurrentLetter = uniqueLettersMap.get(currentLetter);
                uniqueLettersMap.replace(currentLetter, repetitionOfCurrentLetter + 1);
            }
        }

        return uniqueLettersMap;
    }

    public static void printLettersRepetition(HashMap<Character, Integer> uniqueLettersMap) {
        for (Map.Entry<Character, Integer> uniqueLetterWithRepetition : uniqueLettersMap.entrySet()) {
            System.out.println("La Letra " + uniqueLetterWithRepetition.getKey() + " se repite "
                    + uniqueLetterWithRepetition.getValue() + " veces.");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese una cadena: ");
        String myString = scanner.nextLine();

        HashMap<Character, Integer> allUniqueLettersRepetition = CharacterRepetition
                .getUniqueLettersRepetition(myString);

        CharacterRepetition.printLettersRepetition(allUniqueLettersRepetition);

        scanner.close();

    }
}

File: src/c/character_repetition.c
#include <stdio.h>

struct Key
{
    char c;
};

struct Value
{
    int frec;
};

struct HashMap
{
    struct Key key;
    struct Value value;
};

int main()
{
    struct HashMap map[100];
    for (size_t i = 0; i < 100; i++)
    {
        map[i].key.c = ' ';
        map[i].value.frec = 0;
    }

    char input[100];
    printf("Ingrese una cadena: ");
    gets(input);

    int i = 0;
    while (input[i] != '\0')
    {
        char character = input[i];
        int index = input[i] % 100;
        map[index].key.c = input[i];
        printf("La letra %c se repite ", map[index].key.c);
        map[index].value.frec = map[index].value.frec + 1;
        printf("%d vez/veces\n", map[index].value.frec);
        i++;
    }

    return 0;
}